### PR TITLE
Feature/cmisoft 89 Optimizations & fixes

### PR DESCRIPTION
**Pull request by timber-they**
Not much more than the title says; optimizations & fixes.

### CODE REVIEW

File: src/Detectors.jl [hunk 1]
```diff
@@ -30,10 +30,13 @@ Returns a vector of all hits on `detector` for all trajectories in a `solution::
 Uses `add_hits!` internally to achieve this.
 """
 function calculate_hits(
-    # oh boy - is there a better (and still "API-stable") way to pull out the Eltype?!
-    solution::EnsembleSolution{A,B,Vector{RODESolution{C,D,Vector{Eltype},E,F,G,H,I,J,K,L}}},
+    solution::EnsembleSolution{A,B,Vector{C}},
```
Comment (peravali0812) [inline src/Detectors.jl:39]: why?

File: src/Experiment.jl [hunk 2]
```diff
@@ -146,7 +146,11 @@ Simulates an `Experiment`, returning a tuple of `(solution, detector_hits, parti
 """
 function simulate(exp::Experiment)
     u0, params, particles, prob_func = make_initial_values(exp.source, exp.fields, exp.n_particles)
-    prob = SDEProblem{true}(f!, g!, u0, exp.time_span, params)
+    prob = ODEProblem{true}(f!, u0, exp.time_span, params)
+    if (exp.solver == EulerHeun())
```
Comment (peravali0812) [inline src/Experiment.jl:162]: is it for example? there could be other sde solvers

**Review by peravali0812 — APPROVED** [review]

File: src/Detectors.jl (hunk 1 above)
Comment (timber-they) [reply src/Detectors.jl:39]: Because otherwise only SDE solutions are possible.

File: src/Experiment.jl (hunk 2 above)
Comment (timber-they) [reply src/Experiment.jl:162]: For now the only SDE solver supported is `EulerHeun`. If, at some later point, more SDE solvers are supported, this should probably be rewritten to determine the kind of solver used in a better way.

**Pull request merged**